libs/server/Storage/Functions/MainStore/RMWMethods.cs

```diff
@@ -292,23 +292,41 @@ private bool InPlaceUpdaterWorker(ref SpanByte key, ref SpanByte input, ref Span
                     return true;
 
                 case RespCommand.INCR:
-                    long val = NumUtils.BytesToLong(value.AsSpan());
-                    val++;
+                    // Check if value contains a valid number
+                    if (!IsValidNumber(value.LengthWithoutMetadata, value.ToPointer(), ref output, out var val))
+                        return true;
+                    var old = val++;
                     return InPlaceUpdateNumber(val, ref value, ref output, ref rmwInfo, ref recordInfo);
 
                 case RespCommand.DECR:
-                    val = NumUtils.BytesToLong(value.AsSpan());
+                    // Check if value contains a valid number
+                    if (!IsValidNumber(value.LengthWithoutMetadata, value.ToPointer(), ref output, out val))
+                        return true;
                     val--;
                     return InPlaceUpdateNumber(val, ref value, ref output, ref rmwInfo, ref recordInfo);
 
                 case RespCommand.INCRBY:
-                    val = NumUtils.BytesToLong(value.LengthWithoutMetadata, value.ToPointer());
-                    val += NumUtils.BytesToLong(input.LengthWithoutMetadata - RespInputHeader.Size, inputPtr + RespInputHeader.Size);
+                    // Check if value contains a valid number
+                    if (!IsValidNumber(value.LengthWithoutMetadata, value.ToPointer(), ref output, out val))
+                        return true;
+
+                    // Check if input contains a valid number
+                    if (!IsValidNumber(input.LengthWithoutMetadata - RespInputHeader.Size, inputPtr + RespInputHeader.Size, ref output, out var by))
```

Comment:
>!IsValidNumber(input.LengthWithoutMetadata - RespInputHeader.Size, inputPtr + RespInputHeader.Size, ref output, out var by)) [](http://example.com/codeflow?start=24&length=124)

Input validation could be performed in NetworkIncrement.

Reply:
The input is not parsed into a long at the network level. This means that performing the validity check there will force us to parse twice, once for validity check (network level) and second time at IFunctions to perform the corresponding operation.
I guess we can change the input format and pass an 8-byte parsed long value, but this is not the point of the PR and don't know what implications will have for the GarnetApi layer.